single-line-diagram-core/src/main/java/com/powsybl/sld/svg/FeederInfo.java

```diff
@@ -49,7 +51,14 @@ public FeederInfo(String componentType, double value) {
     }
 
     public FeederInfo(String componentType, double value, int precision, String userDefinedId) {
-        this(componentType, value > 0 ? LabelDirection.OUT : LabelDirection.IN, null, precision == 0 || Double.isNaN(value) ? String.valueOf(Math.round(value)) : String.valueOf(Precision.round(value, precision)), userDefinedId);
+        this(componentType, value > 0 ? LabelDirection.OUT : LabelDirection.IN, null, format(value, precision), userDefinedId);
+    }
+
+    private static String format(double value, int precision) {
+        // build pattern
+        String pattern = precision == 0 ? "#;#" : "#.0" + "#".repeat(precision - 1);
+        String formattedValue = new DecimalFormat(pattern, DecimalFormatSymbols.getInstance(Locale.ENGLISH)).format(value);
```

Comment:
To avoid building the pattern I think you could use something like `new DecimalFormat().setDecimalFormatSymbols().setMaximumFractionDigits(precision)`